### ISSUE TITLE
Updated to spring 3.3.4 the same problem

### ISSUE THREAD

**Issue by RamsesGomez** [spring-projects/spring-data-commons#3158]
I just updated to spring 3.3.4 and I'm still having the exact same problem. Am I missing something?

**Comment by RamsesGomez** [comment]
[projections 2.zip](https://github.com/user-attachments/files/17070078/projections.2.zip)

**Comment by mp911de** [comment]
This is correct and expected. Your entity doesn't define an `id` property. The name of the Kotlin property is `_id`. The fact that you're returning `_id` in `getId()` is your choice and we cannot specialize to an accessor that doesn't match the property description.

**Comment by RamsesGomez** [comment]
Well it worked before somehow the update broke it

**Comment by mp911de** [comment]
> how you can give me some suggestion how to handle this correctly, this is a ver common pattern in koltin to hide internals.

When creating entities, the purpose is to interchange data. Deviating property names from how they are exposed adds to complexity and breaks conventions such as in this case.

**Comment by RamsesGomez** [comment]
So why this is broken with the new update I don't understand. What has changed?

**Comment by RamsesGomez** [comment]
I thought entities have to comply with getter/setters in java and there is a getId getter for the Id so I do not understand what the problem is.
